Neos.Flow/Tests/Unit/Http/Component/SessionRequestComponentTest.php

```diff
@@ -0,0 +1,216 @@
+    {
+        $this->mockHttpRequest->method('getCookieParams')->willReturn([
+            'session_cookie_name' => 'some_value',
+        ]);
+
+        $this->mockSessionManager->expects($this->once())->method('initializeCurrentSessionFromCookie')->willReturnCallback(static function (Cookie $cookie) {
+            self::assertSame('session_cookie_name', $cookie->getName());
+        });
+
+        $this->sessionRequestComponent->handle($this->mockComponentContext);
+    }
+
+    public function sessionCookieSettingsProvider(): array
+    {
+        return [
+            ['sessionCookieSettings' => [], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Path=/; HttpOnly'],
+            ['sessionCookieSettings' => ['lifetime' => 123], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Max-Age=123; Path=/; HttpOnly'],
+            ['sessionCookieSettings' => ['path' => '/some/path'], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Path=/some/path; HttpOnly'],
+            ['sessionCookieSettings' => ['secure' => true], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Path=/; Secure; HttpOnly'],
+            ['sessionCookieSettings' => ['httponly' => false], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Path=/'],
+            ['sessionCookieSettings' => ['domain' => 'neos.io'], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Domain=neos.io; Path=/; HttpOnly'],
+            ['sessionCookieSettings' => ['samesite' => 'none'], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Path=/; Secure; HttpOnly; SameSite=none'],
+            ['sessionCookieSettings' => ['samesite' => 'strict'], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Path=/; HttpOnly; SameSite=strict'],
+            ['sessionCookieSettings' => ['samesite' => 'lax'], 'expectedNewCookieValue' => 'session_cookie_name=session-id; Path=/; HttpOnly; SameSite=lax'],
+        ];
+    }
+
+    /**
+     * @test
+     * @dataProvider sessionCookieSettingsProvider
+     */
+    public function newSessionCookiesTakeSessionCookieSettingsIntoAccount(array $sessionCookieSettings, string $expectedCookie): void
+    {
+        $this->mockHttpRequest->method('getCookieParams')->willReturn(['session_cookie_name' => 'session-id']);
+
+        $this->inject($this->sessionRequestComponent, 'sessionSettings', [
+            'name' => 'session_cookie_name',
+            'cookie' => array_merge($this->defaultSessionCookieSettings, $sessionCookieSettings),
+        ]);
+
+        $this->mockSessionManager->expects($this->once())->method('initializeCurrentSessionFromCookie')->willReturnCallback(static function (Cookie $cookie) use ($expectedCookie) {
+            self::assertSame($expectedCookie, (string)$cookie);
+        });
+
+        $this->sessionRequestComponent->handle($this->mockComponentContext);
+    }
+
+    public function cookieValueDataProvider(): array
+    {
+        return [
```

Comment:
Those are the actual test cases that test the bugfix. The other test cases were added boy-scout style :)